libraries/chain/db_maint.cpp

```diff
@@ -925,7 +925,29 @@ void database::perform_chain_maintenance(const signed_block& next_block, const g
    // Reset all BitAsset force settlement volumes to zero
    for( const auto& d : get_index_type<asset_bitasset_data_index>().indices() )
    {
-      modify( d, [](asset_bitasset_data_object& o) { o.force_settled_volume = 0; });
+      if(head_block_time() < HARDFORK_CORE_518_TIME)
+         modify( d, [](asset_bitasset_data_object& o) { o.force_settled_volume = 0; });
+      else { // clean expired feeds
+         modify( d, [this](asset_bitasset_data_object& o) {
+            o.force_settled_volume = 0;
+
+            // Check if asset is smartcoin
+            const auto& settlement_price = o.feeds.begin()->second.second.settlement_price;
+            if(!settlement_price.is_null()) {
+               const auto &asset = get(settlement_price.base.asset_id);
+               auto flags = asset.options.flags;
+               if ((flags & witness_fed_asset) || (flags & committee_fed_asset)) { // if smartcoin
+                  for (auto itr = o.feeds.begin(); itr != o.feeds.end();) { // loop feeds
+                     auto feed_time = itr->second.first;
+                     if (feed_time + (o.options.feed_lifetime_sec) < head_block_time())
```

Comment:
For better performance, please calculate `head_block_time()-feed_lifetime_sec` out of the `for` loop (and be careful about Integer overflow or underflow) and only check `feed_time` while looping.

Reply:
done this one, checked for overflow only as we are dealing with all unsigned ints so underflow will not be possible. overflow could occur with a feed_lifetime_sec is too big. hard to happen in practice as when we are here we are in a smartcoin but check was added anyways.